patches/roslyn-analyzers/0001-WIP-for-source-build.patch

```diff
@@ -0,0 +1,1087 @@
+       <AnalyzerNupkgFile Include="$(PackageTargetsFileDir)\$(PackageTargetsFileName)"/>
+@@ -132,6 +137,8 @@
+       <_CscToolPath Condition="!HasTrailingSlash('$(_CscToolPath)')">$(_CscToolPath)\</_CscToolPath>
+     </PropertyGroup>
+ 
+-    <Exec Command='"$(_CscToolPath)csi.exe" "$(RepoRoot)eng\GenerateAnalyzerNuspec.csx" "$(NuspecFile)" "$(AssetsDir)\" "$(MSBuildProjectDirectory)" "$(Configuration)" "$(TargetFrameworksForPackage)" "@(_NuspecMetadata)" "@(AnalyzerNupkgFile)" "@(AnalyzerNupkgFolder)" "@(AnalyzerNupkgAssembly)" "@(AnalyzerNupkgDependency)" "@(AnalyzerNupkgLibrary)" "$(_GeneratedRulesetsDir)" "$(_GeneratedEditorconfigsDir)" "@(AnalyzerLegacyRuleset)" "$(ArtifactsBinDir)\" "$(AnalyzerDocumentationFileDir)" "$(AnalyzerDocumentationFileName)" "$(AnalyzerSarifFileDir)" "$(AnalyzerSarifFileName)" "$(AnalyzerConfigurationFileDir)" "$(AnalyzerConfigurationFileName)" "$(_GeneratedGlobalAnalyzerConfigsDir)"' />
++    <Exec Command='"$(DotNetExecutible)" "build" "-c" "Release" "$(RepoRoot)eng\GenerateAnalyzerNuspec\GenerateAnalyzerNuspec.csproj"'
++          EnvironmentVariables="MSBUILDDISABLENODEREUSE=1;UseSharedCompilation=false;UseRazorBuildServer=false"/>
++    <Exec Command='"$(DotNetExecutible)" "$(RepoRoot)\artifacts\bin\GenerateAnalyzerNuspec\Release\net5.0\GenerateAnalyzerNuspec.dll" "$(NuspecFile)" "$(AssetsDir)" "$(MSBuildProjectDirectory)" "$(Configuration)" "$(TargetFrameworksForPackage)" "@(_NuspecMetadata)" "@(AnalyzerNupkgFile)" "@(AnalyzerNupkgFolder)" "@(AnalyzerNupkgAssembly)" "@(AnalyzerNupkgDependency)" "@(AnalyzerNupkgLibrary)" "$(_GeneratedRulesetsDir)" "$(_GeneratedEditorconfigsDir)" "@(AnalyzerLegacyRuleset)" "$(ArtifactsBinDir)" "$(AnalyzerDocumentationFileDir)" "$(AnalyzerDocumentationFileName)" "$(AnalyzerSarifFileDir)" "$(AnalyzerSarifFileName)" "$(AnalyzerConfigurationFileDir)" "$(AnalyzerConfigurationFileName)" "$(_GeneratedGlobalAnalyzerConfigsDir)"' />
+   </Target>
+-</Project>
+\ No newline at end of file
++</Project>
+diff --git a/eng/GenerateAnalyzerNuspec/GenerateAnalyzerNuspec.csproj b/eng/GenerateAnalyzerNuspec/GenerateAnalyzerNuspec.csproj
+new file mode 100644
+index 000000000..208270428
+--- /dev/null
++++ b/eng/GenerateAnalyzerNuspec/GenerateAnalyzerNuspec.csproj
+@@ -0,0 +1,8 @@
++<Project Sdk="Microsoft.NET.Sdk">
++
++  <PropertyGroup>
++    <OutputType>Exe</OutputType>
++    <TargetFramework>net5.0</TargetFramework>
++  </PropertyGroup>
++
++</Project>
+diff --git a/eng/GenerateAnalyzerNuspec/Program.cs b/eng/GenerateAnalyzerNuspec/Program.cs
+new file mode 100644
+index 000000000..3c31e343b
+--- /dev/null
++++ b/eng/GenerateAnalyzerNuspec/Program.cs
+@@ -0,0 +1,298 @@
++using System;
++using System.Collections.Generic;
++using System.Diagnostics;
++using System.IO;
++using System.Linq;
++using System.Text;
++
++namespace GenerateAnalyzerNuspec
++{
++    public class Program
++    {
++        public static void Main(string[] Args)
++        {
++            string nuspecFile = Args[0];
```

Comment:
If you leave the original code unindented, would Git treat this as a file move + modify to be a bit more minimal? (And show if you had to make any tweaks?)

Reply:
Tried it out, doesn't seem to help. `git log -p -C` and `git show -C` don't show a diff.

The actual diff is: https://gist.github.com/omajid/e081771153fe7ea0fd4f23917a347e95

Reply:
Huh, I remember reading there's a heuristic with some percentage (90%?) of the file having to be the same to count as a copy/move. Oh well, this is fine. Thanks for the diff.